File: Cap 10/Topico 10.5/Troca Valores V1/tricaval1.pas
{*
 * Livro .....: Algoritmos
 *              Logica para Desenvolvimento de Programacao de
 *              Computadores
 * Autores ...: Jose Augusto N. G. Manzano
 *              Jayr Figueiredo de Oliveira
 * Editora ...: Erica
 * 
 * Capitulo ..: 10 - Topico 10.5
 * 
 * Programa ..: TROCA: CONVENCIONAL
 *}

program TROCA_VALORES_V1;
var
  X, A, B: integer;
begin
  writeln('TROCA DE VALORES - V1');
  writeln;
  write('Entre valor para variavel <A>: ');
  readln(A);
  write('Entre valor para variavel <B>: ');
  readln(B);
  X := A;
  A := B;
  B := X;
  writeln;
  writeln('Os valores trocados sao:');
  writeln;
  writeln('<A> = ', A);
  writeln('<B> = ', B);
end.

File: Cap 10/Topico 10.5/Troca Valores V1/tricaval1.go
/*
 * Livro .....: Algoritmos
 *              Logica para Desenvolvimento de Programacao de
 *              Computadores
 * Autores ...: Jose Augusto N. G. Manzano
 *              Jayr Figueiredo de Oliveira
 * Editora ...: Erica
 *
 * Capitulo ..: 10 - Topico 10.5
 * 
 * Programa ..: TROCA: CONVENCIONAL
 */

package main

import (
  "fmt"
)

var (
  X, A, B int
)

func main() {

  fmt.Println("TROCA DE VALORES - V1")
  fmt.Println()
  
  fmt.Print("Entre valor para variavel <A>: ")
  fmt.Scanln(&A)
  
  fmt.Print("Entre valor para variavel <B>: ")
  fmt.Scanln(&B)
  
  X = A
  A = B
  B = X
  
  fmt.Println()
  fmt.Println("Os valores trocados sao:")
  fmt.Println()
  fmt.Println("<A> =", A)
  fmt.Println("<B> =", B)
  
}
